### ISSUE TITLE
Span passed to IScopeManager.Activate is not == the underlying tracer span type

### ISSUE THREAD

**Issue by ndrwrbgs** [opentracing-contrib/csharp-decorators#8]
Consider
```C#
// ...
IScopeManager.Activate(ISpan span, bool finishSpanOnClose) {
  var myTypedSpan = span as MySpan; // The OpenTracing spec people told me to do this :( I told them it was finicky, as we now see
  // Do stuff
}
// ...
```

Currently, when ScopeBuilderDecorator.StartActive calls `ScopeManagerDecorator.Activate` it does so with the result of `SpanBuilderDecorator.Start()` which is of type `SpanDecorator`.

`ScopeManagerDecorator.Activate` then passes this to `this._scopeManager.Activate` directly

**Comment by jrouaix** [comment]
Right but I don't see why it'll be an issue ?
You really need a span implementation of your own ?
The problem is i'm very glad not to have any IScope overriding to do. And for that I need to give my own Span to the underlying Tracer/Scope

**Comment by ndrwrbgs** [comment]
It's been a long time, but the reason we had to do the casting was one of two things
* We had a strongly typed SpanId struct that could be parsed from a string, but that was wasteful to do on every SetTag/Log/Finish operation, so we tried to save it as a property. I think now I'd end up doing this via baggage though
* There was some problem with IScope not exposing it's `finishSpanOnClose` that prevented another similar wrapping implementation from properly being able to handle `IScopeManager.Active`, since it couldn't tell if the `underlyingManager.Active`'s scope had been set like that. I suggested that `IScope` expose `finishSpanOnClose` since it has to be constructed with one so it makes sense for wrapping, but I know at that time I was told to save it to a property myself and cast the span when it is read.

I think, until I hit whatever case it was that caused me issues again, we can ignore this issue. Especially since, really, if you do need to cast to the concrete type and you're not doing something _crazy_ (like just trying to wrap) then maybe with more people involved it'll be easier to get OT to make a change to that API to expose that property :)
